src/test/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolderTest.java

```diff
@@ -481,6 +482,23 @@ public void deleteWhileComputing() throws Exception {
         assertThat(computation.getResult(), is(Result.ABORTED));
     }
 
+    @Test
+    public void renameWhileComputing() throws Exception {
+        CoordinatedComputedFolder d = r.jenkins.createProject(CoordinatedComputedFolder.class, "d");
+        d.kids.addAll(Arrays.asList("A", "B"));
+        QueueTaskFuture<Queue.Executable> future = d.scheduleBuild2(0).getFuture();
+        FolderComputation<FreeStyleProject> computation;
+        while (Executor.of((computation = d.getComputation())) == null) {
+            Thread.sleep(50);
+        }
```

Comment:
I am not sure this will work everytime. In such cases I prefer to foirce the sync my self. You look at the test in https://github.com/jenkinsci-cert/jenkins/pull/157/files#diff-a686e6a07d22a2e89e35edf7c3257853R194. I use the method `waitForStart` that wait for a build to start before going on.

Secondly, I put a semaphore on a custom build in order to determine myself when the build finishes exactly, so you could try to rename it during the job, receive the error then re-try at the end to receive the "success".

Reply:
The `CoordinatedComputedFolder` here uses a `CountDownLatch` in the [method](https://github.com/jenkinsci/cloudbees-folder-plugin/blob/7e6294bc48d4cddc89c129ceaf1ab6d352462425/src/test/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolderTest.java#L891) that will end up being run by `d.scheduleBuild2(0)`, which will block until [d.onKid("blah")](https://github.com/jenkinsci/cloudbees-folder-plugin/blob/7e6294bc48d4cddc89c129ceaf1ab6d352462425/src/test/java/com/cloudbees/hudson/plugins/folder/computed/ComputedFolderTest.java#L923) is called, so I don't think there is an issue there.

I copied from other tests that actually needed `computation` for what they were testing, but I will change this test to use `waitForStart` instead, and retry checking the name after the computation finishes.